sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -71,7 +71,8 @@ case class DropTable(
     }
     hiveContext.invalidateTable(tableName)
     hiveContext.runSqlHive(s"DROP TABLE $ifExistsClause$tableName")
-    hiveContext.sessionState.catalog.unregisterTable(TableIdentifier(tableName))
+    hiveContext.sessionState.catalog.dropTable(
+      TableIdentifier(tableName), ignoreIfNotExists = true)
```

Comment:
@andrewor14 @yhuai Sorry, I have a question about this. 

When doing `Drop Database`, the changes in my PR affect both `SQLContext` and `HiveContext`. After reading this, I think we need to call `runSqlHive` when the target is Hive tables. Is my understanding is right? Thank you!